api/host.go

```diff
@@ -132,6 +133,18 @@ func ModifyHostStatus(ctx context.Context, env evergreen.Environment, h *host.Ho
 		currentStatus == evergreen.HostQuarantined &&
 		utility.StringSliceContains([]string{evergreen.HostRunning, evergreen.HostProvisioning}, newStatus)
 	if unquarantinedAndNeedsReprovision {
+		query := host.NeedsAgentMonitorDeploy(time.Now())
+		query[host.IdKey] = h.Id
+		foundHost, err := host.FindOne(ctx, query)
+		if err != nil {
+			return "", http.StatusInternalServerError, errors.Wrap(err, HostUpdateError)
+		}
+		if foundHost != nil {
+			if err = h.SetNeedsNewAgentMonitor(ctx, true); err != nil {
+				return "", http.StatusInternalServerError, errors.Wrap(err, HostUpdateError)
+			}
+		}
```

Comment:
This could increase the amount of jobs that are racing because if the agent monitor cron runs a job for this host before the reprovisioning job logic starts on L148, the agent monitor job can run concurrently with the reprovisioning job. The agent monitor shouldn't be deployed until the reprovisioning job fixes the unquarantined host. (The order of operations is unquarantine -> reprovision -> deploy the agent monitor.)

Could we instead modify the host monitoring job to leave the host alone if it needs reprovisioning and its status is running? If it has the reprovision flag, it'll eventually reprovision, so the host monitoring job shouldn't intervene.